static/js/components/forms/B2BPurchaseForm.js

```diff
@@ -166,6 +173,7 @@ class B2BPurchaseForm extends React.Component<Props> {
               <div className="coupon-input-container">
                 <Field type="text" name="coupon" />
                 <button
+                  id="apply-coupon"
```

Comment:
Please also remove this id later, using id is dangerous. As this is a component and if someone use it somewhere else later in project then react will not prevent you and your DOM contains multiple inputs with same id.

Reply:
It's our responsibility to use unique id's for each element.

Reply:
Let me try the refs if that works out.